src/core/createInstance.js

```diff
@@ -43,48 +42,59 @@ export default (
     }
     const config = createConfig(options);
 
-    try {
-      componentRegistry = initializeComponents(config);
-    } catch (e) {
-      configurationFailed = true;
-      throw e;
-    }
+    return initializeComponents(config);
   };
 
   const executeCommand = (commandName, options) => {
-    let command;
+    let execute;
 
-    if (configurationFailed) {
-      // We've decided if configuration fails we'll return
-      // never-resolved promises for all subsequent commands rather than
-      // throwing an error for each of them.
-      command = () => new Promise(() => {});
-    } else if (commandName === "configure") {
-      if (componentRegistry) {
+    if (commandName === "configure") {
+      if (configurePromise) {
         throw new Error(
           "The library has already been configured and may only be configured once."
         );
       }
-      command = configureCommand;
-    } else if (commandName === "log") {
-      command = logCommand;
+
+      execute = () => {
+        configurePromise = configureCommand(options);
+        return configurePromise;
+      };
     } else {
-      if (!componentRegistry) {
+      if (!configurePromise) {
         throw new Error(
           `The library must be configured first. Please do so by calling ${namespace}("configure", {...}).`
         );
       }
 
-      command = componentRegistry.getCommand(commandName);
-
-      if (!isFunction(command)) {
-        throw new Error(`The ${commandName} command does not exist.`);
+      if (commandName === "log") {
+        execute = () => logCommand(options);
+      } else {
+        execute = () => {
+          return configurePromise.then(
```

Comment:
By waiting for `configurePromise` to complete before executing any other commands, I was able to remove the crufty workaround that was in `createLifecycle.js`.